### PR TITLE
First pass -- phase 1, extract manifest as dict, and phase 2, convert into topic tree

### PR DESCRIPTION
**Pull request by divad12**
Works for phase 1 and 2 as described here:

https://www.notion.so/learningequality/IMS-package-Kolibri-node-s-converter-is-created-7516609da73a4f439bb0b5f9be36c108

Tested only for http://procomun.educalab.es/es/ode/view/1465806119010 for web content nodes.

#  Test plan

1. Download sample IMS package from http://procomun.educalab.es/es/ode/view/1465806119010 (find the dropdown on the top right of the page content, and select IMSCP from the list [direct link](https://agrega.educacion.es///export/es_2016062312_9100647/IMS_CP/Evento's_Solutions,_servicios_integrales_(ESSI)-IMS_CP.zip)), into the same directory as this project.
1. Uncomment `#preview_in_browser` in `ricecooker_utils.py`
1. Create a file and insert the following contents:
```
#!/usr/bin/env python

import logging
import tempfile

from ricecooker.classes import licenses

from imscp import extract_from_zip
from ricecooker_utils import make_topic_tree


if __name__ == '__main__':
    license = licenses.SpecialPermissionsLicense(
        description="XXX",
        copyright_holder="XXXX"
    )
    logging.basicConfig(level=logging.DEBUG)
    with tempfile.TemporaryDirectory() as extract_path:
        imscp_dict = extract_from_zip('eventos.zip', license, extract_path)
        for topic_dict in imscp_dict:
            topic_tree = make_topic_tree(license, topic_dict)
            print('topic tree', topic_tree)

```
1. Run it.
1. See the following

image

1. Comment out `preview_in_browser` in `ricecooker_utils.py`
1. Run the test runner above again, and see the following output:

```
INFO:root:Extracting zip file eventos.zip to /var/folders/z8/9d4m1_l55wl7q0ff6shs8c8w0000gn/T/tmpwp4uazxg
INFO:root:Parsing imsmanifest.xml in /var/folders/z8/9d4m1_l55wl7q0ff6shs8c8w0000gn/T/tmpwp4uazxg
INFO:root:Extracting tree structure ...
INFO:root:
topic tree Evento's Solutions, servicios integrales (ESSI) (TopicNode): 6 descendants
```

### CODE REVIEW

File: imscp.py
Comment (kollivier) [inline imscp.py]: Actually, thinking more about this, let's move all the HTML5 zip creation work to phase 2, when we add ricecooker packaging support. We will need the zips regardless of how we package, but I think it might be best to keep the parsing code focused solely on parsing, and do any transformations as a later step.

For now, let's just return the path to the actual file(s) so that it's up to the consumer of that data to decide what to do with it. Maybe go ahead and create a `ricecooker_utils.py` or something and put this function in there for now.

File: imscp.py
Comment (kollivier) [inline imscp.py]: Minor nitpick, but could we switch these lines to use the `logging` module instead since this will ultimately be used as a library? `__main__` can call `logging.basicConfig` and set it up to print to the console so that the behavior is the same.

File: imscp.py
Comment (kollivier) [inline imscp.py]: Once we move the zip packaging code out of this step, we can remove this as well.

**Review by kollivier — CHANGES_REQUESTED** [review]
Overall, looks really good! The HTML5 zip change is the only real thing we need to change, and that was my bad for not thinking this through well enough!

File: imscp.py
Comment (divad12) [reply imscp.py]: Done

File: imscp.py
Comment (divad12) [reply imscp.py]: done!

File: imscp.py
Comment (divad12) [reply imscp.py]: done

**Review by kollivier — APPROVED** [review]
Approved, but would like if we could have some basic docs on function and params before merging!

**Comment by divad12** [comment]
Added docs to `readme.md` and example chef script to `examples/`

**Review by kollivier — APPROVED** [review]

**Pull request merged**
